torchtune/modules/transformer.py

```diff
@@ -189,12 +202,19 @@ def forward(self, tokens: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
         # shape: [b, s, d]
         h = self.tok_embeddings(tokens)
 
-        mask = None
+        # shape: [b, 1, s, s]
+        if mask is not None:
+            mask = mask[:, None, :, :]
```

Comment:
Thoughts on doing this in attention.py? Cause (a) that's where we actually need it to be this shape, and (b) then we do not have separate contracts on mask shape in `TransformerDecoder` and `TransformerDecoderLayer`

Reply:
yes, I think that makes sense